### PR TITLE
Update NDK from r21 to r22

### PR DESCRIPTION
**Pull request by bdero**
Resolves https://github.com/flutter/flutter/issues/73818.

All of the platform-specific binaries were consolidated under `toolchains/llvm/sysroot` subdirs. The unified sysroot is well-structured in terms of headers, but there are no longer separate well-structured sysroots for each architecture in terms of binary placement, so I opted to just include the correct library directory at link time.

RIP MIPS. I removed some of the MIPS stuff that was in the way, but in order to keep the diff relevant, I'll go through and remove the rest of it in a follow-up. Note that MIPS was removed in NDK r17.

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM assuming everything still compiles :)

@blasten @jason-simmons fyi

**Comment by bdero** [comment]
Working out a remaining path problem with the 32 bit backends.

**Review by blasten — APPROVED** [review]
Thanks for the clean up!

**Comment by chinmaygarde** [comment]
Can we land and roll this?

**Comment by bdero** [comment]
This should be good to go by EOD today or tomorrow morning. Aarch64/arm/x64 targets are good to go, but my attention got a bit split while setting up an environment to test x86 (mac toolchain doesn't have an x86 backend).

**Comment by chinmaygarde** [comment]
I'd just let the presubmits test it. But your call.

**Comment by dnfield** [comment]
We can always revert if it breaks on post. It's generally hard to capture all the variants locally when doing rolls like this.

**Comment by bdero** [comment]
Yeah that's true. I'm probably being too cautious here. Merging it.

**Pull request merged**
